#!/bin/bash
#PBS -l select=1:ncpus=4:mem=40GB
#PBS -l walltime=7:00:00

module load strelka/2.9.9
module load bcftools
temporaryDir=${projectDir/project/scratch/}

echo "Time: $(date). Begin germline Single Nucleotide Variant detection for $SID." >> $projectDir/DNAlog.txt
if [ -d $temporaryDir/g/$SID/ ] # Strelka will throw an error. Directory must be empty.
then
  rm -r $temporaryDir/g/$SID/
fi
configureStrelkaGermlineWorkflow.py --referenceFasta=/project/HeadNeck/sequence/hg38/GCA_000001405.15_GRCh38_no_alt_analysis_set.fna --bam=$normalFile --runDir=$temporaryDir/g/$SID/
$temporaryDir/g/$SID/runWorkflow.py -m local -j 4
rename S1.vcf $SID.vcf $temporaryDir/g/$SID/results/variants/* # File output by Strelka always has S1 in its name. User can't specify results output name.
mv $temporaryDir/g/$SID/results/variants/genome.$SID* $projectDir/singleNucleotideVariants/
bcftools view -f PASS -o $projectDir/singleNucleotideVariants/genome.$SID.candidates.vcf.gz -O z $projectDir/singleNucleotideVariants/genome.$SID.vcf.gz
bcftools index -t $projectDir/singleNucleotideVariants/genome.$SID.candidates.vcf.gz
echo "Time: $(date). Complete germline Single Nucleotide Variant detection for $SID." >> $projectDir/DNAlog.txt
