#!/bin/bash

#PBS -l ncpus=1
#PBS -l mem=30GB
#PBS -l walltime=0:50:00
#PBS -l storage=scratch/hm82

module load R/3.6.1
module load java
module load bwa

genome=/scratch/hm82/Reference/hs38DH.fasta
BSgenome=BSgenome.Hsapiens.UCSC.hg38

SVpath=$projectDir/structuralVariants/${outputID}structural.vcf.gz # Replace mapped directory by variants directory in path.
somaticSVpath=${SVpath/.vcf/Somatic.vcf}
retroSVpath=${SVpath/.vcf.gz/AnnotatedRetro.vcf} # Annotated with Retrotransposons.
retroViralSVpath=${retroSVpath/AnnotatedRetro/AnnotatedRetroViral} # Annotated with Viruses.

echo "Time: $(date). Begin structural variant filtering and annotation for $outputID" >> $projectDir/DNAlog.txt
Rscript /home/562/ds6924/software/gridss_somatic_filter.R --pondir $projectDir/structuralVariants/ --ref $BSgenome --input $SVpath --output $somaticSVpath --scriptdir /home/562/ds6924/software
java -Xmx26g -cp /home/562/ds6924/software/gridss.jar gridss.AnnotateInsertedSequence THREADS=1 REFERENCE_SEQUENCE=$genome INPUT=$somaticSVpath OUTPUT=$retroSVpath ALIGNMENT=REPLACE REPEAT_MASKER_BED=/home/562/ds6924/databases/hg38repeats.bed
java -Xmx26g -cp /home/562/ds6924/software/gridss.jar gridss.AnnotateInsertedSequence THREADS=1 REFERENCE_SEQUENCE=/home/562/ds6924/sequences/human_virus.fa ALIGNMENT=APPEND INPUT=$retroSVpath OUTPUT=$retroViralSVpath
echo "Time: $(date). Complete structural variant filtering and annotation for $outputID." >> $projectDir/DNAlog.txt
